agent_secret_manager/src/subject.rs

```diff
@@ -50,8 +51,21 @@ impl Sign for Subject {
     async fn key_id(&self, subject_syntax_type: &str) -> Option<String> {
         let method: DidMethod = serde_json::from_str(&format!("{subject_syntax_type:?}")).ok()?;
 
+        if method == DidMethod::Web {
+            return self
+                .secret_manager
+                .produce_document(
+                    method,
+                    Some(did_manager::MethodSpecificParameters::Web { origin: origin() }),
+                )
+                .await
+                .ok()
+                .and_then(|document| document.verification_method().first().cloned())
+                .map(|first| first.id().to_string());
+        }
+
```

Comment:
`[resolved]` 

In order to counter some repeated code you could consider implementing this method like so:
```rust
    async fn identifier(&self, subject_syntax_type: &str) -> anyhow::Result<String> {
        let method: DidMethod = serde_json::from_str(&format!("{subject_syntax_type:?}"))?;

        let method_specific_parameters =
            (method == DidMethod::Web).then(|| did_manager::MethodSpecificParameters::Web { origin: origin() });

        Ok(self
            .secret_manager
            .produce_document(method.clone(), method_specific_parameters)
            .await
            .map(|document| document.id().to_string())?)
    }
```

where
```rust
        let method_specific_parameters =
            (method == DidMethod::Web).then(|| did_manager::MethodSpecificParameters::Web { origin: origin() });
```

will result in `None` for all DID Methods other than did:web

Reply:
Thanks! Will consider it in the follow-up PR 👍